app/rtc_s1.py

```diff
@@ -0,0 +1,1262 @@
+                                             output_file,
+                                             cfg.geogrid, verbose=False)
+            if flag_hdf5:
+                temp_files_list.append(output_file)
+            else:
+                output_file_list.append(output_file)
+
+        if flag_hdf5:
+            if flag_save_nlooks:
+                nlooks_mosaic_file = output_metadata_dict['nlooks'][0]
+            else:
+                nlooks_mosaic_file = None
+            if flag_save_rtc_anf:
+                rtc_anf_mosaic_file = output_metadata_dict['rtc'][0]
+            else:
+                rtc_anf_mosaic_file = None
+
+            # Update metadata datasets that depend on all bursts
+            sensing_start = None
+            sensing_stop = None
+            for burst_id, burst_pol_dict in cfg.bursts.items():
+                pols = list(burst_pol_dict.keys())
+                burst = burst_pol_dict[pols[0]]
+                print('this burst:')
+                if sensing_start is not None:
+                    print('    ', sensing_start.strftime('%Y-%m-%dT%H:%M:%S.%f'))
+                if sensing_stop is not None:
+                    print('    ', sensing_stop.strftime('%Y-%m-%dT%H:%M:%S.%f'))
+                if (sensing_start is None or
+                        burst.sensing_start < sensing_start):
+                    sensing_start = burst.sensing_start
+                    print('updated sensing start')
+                if sensing_stop is None or burst.sensing_stop > sensing_stop:
+                    sensing_stop = burst.sensing_stop
+                    print('updated sensing stop')
+
+            sensing_start_ds = f'{BASE_DS}/identification/zeroDopplerStartTime'
+            sensing_end_ds = f'{BASE_DS}/identification/zeroDopplerEndTime'
+            del hdf5_obj[sensing_start_ds]
+            del hdf5_obj[sensing_end_ds]
+            hdf5_obj[sensing_start_ds] = \
+                sensing_start.strftime('%Y-%m-%dT%H:%M:%S.%f')
+            hdf5_obj[sensing_end_ds] = \
+                sensing_stop.strftime('%Y-%m-%dT%H:%M:%S.%f')
+
+            save_hdf5_file(hdf5_obj, output_hdf5_file, flag_apply_rtc,
+                           clip_max, clip_min, output_radiometry_str,
+                           output_file_list, cfg.geogrid, pol_list,
+                           geo_filename, nlooks_mosaic_file,
+                           rtc_anf_mosaic_file, radar_grid_file_dict)
+
+    logger.info('removing temporary files:')
+    for filename in temp_files_list:
+        if not os.path.isfile(filename):
+            continue
+        os.remove(filename)
+        logger.info(f'    {filename}')
+
+    logger.info('output files:')
+    for filename in output_file_list:
+        logger.info(f'    {filename}')
+
+    t_end = time.time()
+    logger.info(f'elapsed time: {t_end - t_start}')
+
+
+def save_hdf5_file(hdf5_obj, output_hdf5_file, flag_apply_rtc, clip_max,
```

Comment:
You could easily separate these I/O functions from the main run function and put them in a separate file